### ISSUE TITLE
[Bug]: Unable to rotate uploaded ssl certificate with random_pet resource

### ISSUE THREAD

**Issue by garry-t** [hetznercloud/terraform-provider-hcloud#846]
### What happened?

Impossible upload certificate if random_pet resource used

terraform  = 1.2.4
hetznercloud/hcloud = 1.42.1

```
Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to hcloud_uploaded_certificate.certificate2, provider
│ "provider[\"registry.terraform.io/hetznercloud/hcloud\"]" produced an
│ unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

### What did you expect to happen?

It is common case when you can rotate certificate w/o breaking resource functionality. 
Provided example it is already some work around, when you cant update certificate in place. 
Ideally i would like to have ability update certificate in place, w/o declaring second resource.

### Please provide a minimal working example

```terraform
data "local_file" "certificate" {
  filename = "cert_name.crt"
}
data "local_file" "private_key" {
  filename = "private key.key"
}

resource "random_pet" "prefix" {
  keepers = {
    # Generate a new pet name each time we switch to a new ssl
    name = var.hetzner_certificate_name
  }
  separator = "_"
}

resource "hcloud_uploaded_certificate" "certificate2" {
  name        = random_pet.prefix.keepers.name 
  certificate = data.local_file.certificate.content
  private_key = data.local_file.private_key.content
}
```

Working solution 

```terraform
data "local_file" "certificate" {
  filename = "cert_name.crt"
}
data "local_file" "private_key" {
  filename = "private key.key"
}

resource "hcloud_uploaded_certificate" "certificate2" {
  name         = "hardcoded_name"
  certificate = data.local_file.certificate.content
  private_key = data.local_file.private_key.content
}

```

**Comment by jooola** [comment]
Hi @garry-t 

Before trying to debug your issue, have you tried upgrading Terraform and the hcloud provider you are using?

Hashicorp recommends staying current within two (2) releases from the latest major release in order to receive optimal support.  
https://support.hashicorp.com/hc/en-us/articles/360021185113

**Comment by jooola** [comment]
Here is a working example that I created while trying to understand your use case:

```tf
terraform {
  required_providers {
    hcloud = {
      source  = "hetznercloud/hcloud"
      version = "1.45.0"
    }

    random = {
      source  = "hashicorp/random"
      version = "3.6.0"
    }
  }
}

provider "hcloud" {}
provider "random" {}

variable "hetzner_certificate_name" {
  type    = string
  default = "my-certificate"
}

data "local_file" "certificate" {
  filename = "fake.crt"
}
data "local_file" "private_key" {
  filename = "fake.key"
}

resource "random_pet" "prefix" {
  prefix    = var.hetzner_certificate_name
  separator = "_"

  keepers = {
    certificate = sensitive(data.local_file.certificate.content) # Generate a name if the certificate file content changed.
  }
}

resource "hcloud_uploaded_certificate" "certificate" {
  name        = random_pet.prefix.id # Use generated random name e.g. "my-certitificate_spaniel"
  certificate = sensitive(data.local_file.certificate.content)
  private_key = sensitive(data.local_file.private_key.content)
}
```

If the certificate files are updated, terraform will propose the following:

![image](https://github.com/hetznercloud/terraform-provider-hcloud/assets/19195485/827aef27-7319-4d72-8b5f-1306949067c6)

**Comment by garry-t** [comment]
@jooola what will be after you press yes button?
 No I didint try to update any version since we got  this case on production env, it is not possible. But for debugging issue,  I'll try with latest version of terraform and hetzner.

**Comment by jooola** [comment]
The terraform state will then contain the new hcloud certificate with new pet name and new cert file content. The previous hcloud certificate will be destroyed.

![image](https://github.com/hetznercloud/terraform-provider-hcloud/assets/19195485/6ead3662-7673-459c-be8e-a4f02cffed5f)

**Comment by garry-t** [comment]
@jooola ok. Let me reproduce it with latest terraform

**Comment by apricote** [comment]
I did not fully understand what you are trying to achieve, but you might be able to use [`lifecycle.create_before_destroy`](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#create_before_destroy) to create the new certificate before deleting the old one.

**Comment by garry-t** [comment]
use case: 
SSL certificate expire soon and to be replaced for LB.
Scenario:
1. New ssl cert files obtained and placed to the project folder.
2.  Loadbalancer use SSL certificate with name `cert1`.
3. Run tf code which we have to replace ssl certificate in place with option  [lifecycle.create_before_destroy](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#create_before_destroy) Result: fails. Reason: cert name duplication. this is OK. 

Workaround: 
See my original post. I created new resource, so only certificate with random name uploads and exact this step fails.
The upload of certificate didn't work when `random_pet` use. 
I replaced random_pet with string "cert2" and it worked. 
I cannot reproduce it with `terraform 1.6.6` and test env. Next certificate rotation cycle will be rechecked as well on real env.
closed for now.
